src/lib.rs

```diff
@@ -1041,6 +1041,48 @@ impl<T: Clone> Receiver<T> {
             .try_recv_at(&mut self.pos)
             .map(|cow| cow.unwrap_or_else(T::clone))
     }
+
+    /// Produce a clone of this Receiver that has the same messages queued.
+    ///
+    /// The [`Receiver::clone`] method creates a new receiver that starts with zero messages
+    /// available.  In some cases, it is useful to produce a channel that starts at the same point
+    /// as an existing `Receiver` instead of skipping all currently queued messages.
```

Comment:
* Would be great if you provided the description/rationale in the PR and commit descriptions.
* I'd imagine this would apply to all channel API out there. Is there any prior art here?
* Unless there is prior-art we can follow the example of (and if that makes sense), I think a better API would be a separate setting on the receiver to set/unset this behavior on existing `Clone::clone` impl.

Reply:
> I'd imagine this would apply to all channel API out there. Is there any prior art here?

I believe this is one of the reasons `tokio`'s broadcast receiver does not implement `Clone`, but have not looked into the cause.  It really only applies to broadcast channels; for work-queue type channels all receivers are "empty".

> Unless there is prior-art we can follow the example of (and if that makes sense), I think a better API would be a separate setting on the receiver to set/unset this behavior on existing Clone::clone impl.

That seems needlessly complex and very prone to error (you'd either need to set it on all channels, which is prone to forgetting unless you wrap the creation function, or set it prior to clone, which defeats the point of having a trait for cloning).

A better solution might be to deprecate or remove clone (so existing users notice) and provide two new methods to create a `Receiver` given one: `real_clone` (which is a bad name, I'm open to suggestions for a better one) and `new_receiver`.

Or just change how clone works and document it in the API change.  In the common case where clones are made during channel startup, the two are identical, and in many other cases it doesn't matter if you get older messages from a new clone.